subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java

```diff
@@ -340,12 +340,20 @@ public void connect(PipedInputStream snk) throws IOException {
      * also switch to the more specific {@link #expectDocumentedDeprecationWarning(String)} if the warning includes a documentation
      * link and you don't want to (ironically) see code testing deprecation appearing as if it itself were deprecated.
      */
-    GradleExecuter expectDeprecationWarning(String warning);
+    default GradleExecuter expectDeprecationWarning(String warning) {
```

Comment:
I agree we don't need to, but what's the downside?  The `withStdIn` methods in this interface already follow this pattern.  This way requires slightly less code, since we don't have to rewrite the method declaration in the abstract implementation.  I think this is a very common pattern.

Reply:
As long as we're already doing this I guess it's fine.

In general I prefer to leave interfaces as interfaces unless there is some upside to it, for example like 10 implementations of it and only one needs to have a different implementation for a method. Otherwise, the interface turns into more of an implementation itself and we've thrown away the whole concept of implementation vs interface.

An interface with all default methods is pretty much an abstract class

Reply:
This one here I think is pretty benign, but imo things get pretty smelly when I see a lot of code in an interface

Reply:
Agreed.  Providing convenience overloads that just forward calls is a strong use case for defaults, other use cases need to justify themselves a little more.